### PR TITLE
refactor: process user delete event - WPB-10193

### PR DESCRIPTION
**Pull request by jullianm**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10193" title="WPB-10193" target="_blank">TaskWPB-10193</a>  [iOS] Process UserDeleteEvent
  </td></table>
  <br />
 

### Key points

This PR is part of the quick sync refactoring plan and is related to processing the multiple events we receive from the backend or the push channel.

Specifically, this PR is about porting the existing implementation of the `UserDelete` event.

### Testing

UTs cover the following use cases, ensuring that
- the processor correctly invokes the user repository dedicated method `deleteUserAccount`.

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: WireDomain/Sources/WireDomain/Repositories/Team/TeamRepository.swift [hunk 1]
```diff
@@ -103,8 +103,9 @@ final class TeamRepository: TeamRepositoryProtocol {
             return member
         }
 
-        await userRepository.deleteUserAccount(
-            for: user,
+        try await userRepository.deleteUserAccount(
+            with: userID,
+            domain: nil,
```
Comment (netbe) [inline WireDomain/Sources/WireDomain/Repositories/Team/TeamRepository.swift]: don't you have to pass the domain?

File: WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/Event decoding/Team/TeamMemberLeaveEventDecoder.swift [hunk 2]
```diff
@@ -41,7 +41,7 @@ struct TeamMemberLeaveEventDecoder {
         return TeamMemberLeaveEvent(
             teamID: teamID,
             userID: payload.userID,
-            time: time.date
+            time: time?.date ?? .now
```
Comment (netbe) [inline WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/Event decoding/Team/TeamMemberLeaveEventDecoder.swift:44]: question: is the date even used ? or can we keep it optional?

File: WireDomain/Sources/WireDomain/Repositories/Team/TeamRepository.swift (hunk 1 above)
Comment (jullianm) [reply WireDomain/Sources/WireDomain/Repositories/Team/TeamRepository.swift]: fixed in https://github.com/wireapp/wire-ios/pull/2001/commits/7f25694dccd08ea38100bc39c633e764ffad7298

File: WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/Event decoding/Team/TeamMemberLeaveEventDecoder.swift (hunk 2 above)
Comment (jullianm) [reply WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/Event decoding/Team/TeamMemberLeaveEventDecoder.swift:44]: yes it is used when the user is removed from conversations, we append a system message (`func appendSystemMessage` from `ZMConversation`) that requires a non optional date. I've double checked on Android they decode this date from the event as well.

**Review by johnxnguyen — APPROVED** [review]

**Review by netbe — APPROVED** [review]

**Pull request merged**